core/domain/user_services.py

```diff
@@ -582,13 +582,18 @@ def _save_user_settings(user_settings):
     user_model = user_models.UserSettingsModel.get_by_id(user_settings.user_id)
     if user_model is not None:
         user_model.populate(**user_settings_dict)
-        user_model.update_timestamps()
-        user_model.put()
     else:
         user_settings_dict['id'] = user_settings.user_id
-        model = user_models.UserSettingsModel(**user_settings_dict)
-        model.update_timestamps()
-        model.put()
+        user_model = user_models.UserSettingsModel(**user_settings_dict)
+
+    # TODO(#12755): The roles and banned fields of UserSettingsModel is not in
+    # use and expected to be used once it's populated with correct value.
+    # The update_roles_and_banned_fields method helps populate correct value in
+    # the new fileds. It is expected to remove update_roles_and_banned_fields
```

Comment:
```suggestion
    # The update_roles_and_banned_fields call ensures that the new fields are always correct in 
    # the model. It is expected to remove update_roles_and_banned_fields
```

Should I change it to this^? (Or the existing comment is fine?)

Reply:
Maybe `ensures that the new fields always contain correct values. It is expected`

Reply:
Done!